From 8c61a60838d8410740d7312ec4a6e3b6fb52deba Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:40:51 -0400
Subject: [alsa] hda: Fix pop noises on reboot for Dell XPS 13 9333

Message-id: <140972823573-417-patchset-jkysela@redhat.com>
Patchwork-id: 91113
O-Subject: [RHEL 7.1 PATCH 416/442] ALSA: hda - Fix pop noises on reboot for Dell XPS 13 9333
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

If nid 0x15 (Headphone Playback Switch) is in D3 and headphones are
plugged in when the laptop reboots, a pop noise is generated.
Prevent this by keeping nid 0x15 in D0 when headphones are plugged in.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=76611
Signed-off-by: Gabriele Mazzotta <gabriele.mzt@gmail.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: de3da4f696b473e7810bd9837c18bf9fd30a9a6d)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 3ebb761..6b38ec3 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4093,7 +4093,7 @@ static unsigned int alc_power_filter_xps13(struct hda_codec *codec,
 
  /* Avoid pop noises when headphones are plugged in */
  if (spec->gen.hp_jack_present)
-  if (nid == codec->afg || nid == 0x02)
+  if (nid == codec->afg || nid == 0x02 || nid == 0x15)
    return AC_PWRST_D0;
  return power_state;
 }
-- 
1.7.1